haas/model.py

```diff
@@ -40,17 +42,30 @@
 
 db = SQLAlchemy(app)
 
+# Sets up variant type so that postgresql can use BIGINT primary keys
+# while sqlite uses Integer primary keys.
+BigIntegerType = BigInteger().with_variant(
+        sqlite.INTEGER(), 'sqlite')
+
 
 def init_db(uri=None):
     """Start up the DB connection.
 
-    `uri` is the uri to use for the databse. If it is None, the uri from the
+    `uri` is the uri to use for the database. If it is None, the uri from the
     config file will be used.
     """
     if uri is None:
         uri = cfg.get('database', 'uri')
     app.config.update(SQLALCHEMY_DATABASE_URI=uri)
 
+    global BigIntegerType
+
+    # Decide if primary keys should be BIGINTs or Integers
+    if 'postgresql' in uri:
```

Comment:
Given that bigint [should be the same as an int](https://www.sqlite.org/datatype3.html) (since sqlite dynamically sizes the field based on the value), I'm not sure it would hurt anything to use bigint with sqlite and get rid of BigIntegerType completely.

Also, I just tried setting an sqlite PRIMARY key to a bigint, and it seems to work!
```sql
sqlite> create table foo1 ( foo1 int PRIMARY KEY );
sqlite> create table foo2 ( foo1 bigint PRIMARY KEY );
sqlite> 
```